src/include/encoding.h

```diff
@@ -307,6 +307,26 @@ void decode(std::chrono::time_point<Clock, Duration>& t,
   t = Clock::from_timespec(ts);
 }
 
+template<typename Rep, typename Period>
```

Comment:
Looks good.

(Might want to constrain them for Rep being an integral type, just so someone doesn't try to put a double through it.)